### DISCUSSION TITLE
Incremental static regeneration and monthly function usage

### DISCUSSION THREAD

**Discussion by ricokahler** [vercel/next.js#19894] [Help]
Hey all,

I have a question regarding incremental static regeneration and it's relationship to Vercel monthly function usage.

I've tried to look at the code and see how incremental static regeneration is implemented and afaik, it seems like it uses some webpack loaders to generate some serverless functions and then the serverless functions set the [`stale-while-revalidate` cache header](https://vercel.com/docs/edge-network/caching#stale-while-revalidate).

My question is whether or not adding `revalidate: 1` to `getStaticProps` will turn the static into a serverless function:

1. Is this understanding correct? Is it no longer a static page and is instead a serverless function?
2. If so, how does this affect serverless function usage? Does every request to a `revalidate: 1` page also cause a serverless function execution?
3. If so, is there a way to decrease the cost hit? What happens when I increase the revalidate value (e.g. what does `revalidate: 3600` do?)

Note: this question is not necessarily about performance. It's purely around cost and [limits](https://vercel.com/docs/platform/fair-use-policy#typical-monthly-usage-guidelines).

**Comment by shootingsupplies** [comment]
Hi @ricokahler Did you get any info on this?
